crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -149,9 +148,20 @@ impl KitsuneP2pActor {
         let f = if !is_mock {
             let mut conf = kitsune_p2p_proxy::tx2::ProxyConfig::default();
             conf.tuning_params = Some(config.tuning_params.clone());
-            if let Some(use_proxy) = &tx2_conf.use_proxy {
-                let proxy_url = ProxyUrl::from(use_proxy.as_str());
-                conf.client_of_remote_proxy = Some(proxy_url);
+            match tx2_conf.use_proxy {
+                KitsuneP2pTx2ProxyConfig::NoProxy => (),
+                KitsuneP2pTx2ProxyConfig::Specific(proxy_url) => {
+                    conf.client_of_remote_proxy = ProxyRemoteType::Specific(proxy_url);
+                }
+                KitsuneP2pTx2ProxyConfig::Bootstrap {
+                    bootstrap_url,
+                    fallback_proxy_url,
+                } => {
+                    conf.client_of_remote_proxy = ProxyRemoteType::Bootstrap {
+                        bootstrap_url,
+                        fallback_proxy_url,
+                    };
+                }
```

Comment:
Perhaps this is where you want to query the server for a proxy url, and just keep this as an Optional url based on the result?

Reply:
@maackle - I think we need this to be a runtime thing. As network conditions are changing, or we're explicitly removing / adding proxy servers to the proxy_list, I think we want running nodes that get disconnected to re-query and update to keep nodes running, right?